### DISCUSSION TITLE
Camera FOV

### DISCUSSION THREAD

**Discussion by shu-bamma** [larics/icuas22_competition#45] [Q&A] [answered]
@fpetric @lmark1 
We were unable to find the camera plugin in the files. Can you tell us what is the camera FOV that has been simulated ?
Also what camera will be used in the hardware runs?

**Comment by fpetric** [comment, marked as answer]
Hi @shu-bamma 

camera plugin is the standard `gazebo_ros_openni_kinect` plugin from the `gazebo_plugins` package at https://github.com/ros-simulation/gazebo_ros_pkgs.

It is attached to the UAV within the [ardupilot_gazebo](https://github.com/larics/ardupilot_gazebo) package through [kopterworx.urdf.xacro ](https://github.com/larics/ardupilot_gazebo/blob/e8f99b5ebee303ad12bf7ae954ca21585c6ac3b7/models/kopterworx/urdf/kopterworx.urdf.xacro) that builds on top of the [multirotor_base.urdf.xacro](https://github.com/larics/ardupilot_gazebo/blob/e8f99b5ebee303ad12bf7ae954ca21585c6ac3b7/models/util/multirotor_base.urdf.xacro#L230) with the default parameters:
          
          <horizontal_fov>1.3962634</horizontal_fov>

          <image>
          
            <width>640</width>
            
            <height>480</height>
            
            <format>B8G8R8</format>
            
          </image>
          
 The visuals of the camera in gazebo are modeled after Asus Xtion camera, but on the real platform we will be using one of the cameras from the Intel Realsense portfolio (depending on availability with the current issues in supply chains).
